controllers/auth.js

```diff
@@ -0,0 +1,59 @@
+import db from '../models';
+import Auth from '../helpers/auth';
+
+export const signup = async (req, res) => {
+  const {
+    username, firstname, lastname, email, password,
+  } = req.body;
+  const hashedPassword = Auth.hashPassword(password);
+  const transaction = await db.sequelize.transaction();
+  try {
+    const newUser = await db.users.create({
+      username, firstname, lastname, email, password: hashedPassword,
+    }, {
+      transaction,
+    });
+    await transaction.commit();
+    if (newUser) {
+      return res.status(201).json({
+        message: 'created successfully',
+        user: {
+          token: Auth.genToken(username, email),
+          username: newUser.username,
+          email: newUser.email,
+        },
+      });
+    }
+  } catch (ex) {
+    await transaction.rollback();
+    return res.status(409).json({ message: `${ex.errors[0].path.toLowerCase()} already exists` });
+  }
+};
+
+export const signin = async (req, res) => {
+  const { email, password } = req.body;
+  const transaction = await db.sequelize.transaction();
+  try {
+    const user = await db.users.findOne({ where: { email: req.body.email } }, { transaction });
```

Comment:
This users object can be destructured from the db object you imported e.g 

`import db from ../models`;
const { users } = db;